msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByAuthorizationGrantSupplier.java

```diff
@@ -77,55 +77,56 @@ private boolean IsUiRequiredCacheSupported() {
     private OAuthAuthorizationGrant processPasswordGrant(
             OAuthAuthorizationGrant authGrant) throws Exception {
 
-        if (!(authGrant.getAuthorizationGrant() instanceof ResourceOwnerPasswordCredentialsGrant)) {
+        if (!(authGrant.getParameters().get(GrantConstants.GRANT_TYPE_PARAMETER).get(0).equals(GrantConstants.PASSWORD))) {
```

Comment:
issue: see my comments below - the grant should be a strongly typed parameter. It will make the code easier to read and more performant.

Reply:
What other comment are you referring to? Also, by "strongly typed parameter" do you mean:
- An enum with all the grant types?
- A field in OAuthAuthorizationGrant (so it'd be just `authGrant.grantType()`)?
- Specific classes for each grant, like the `ResourceOwnerPasswordCredentialsGrant` that used to be on this line?

Reply:
Agreed with @bgavrilMS. `getParametrs().get(...).get(N)` is a confusing pattern to keep reading. `OAuthAuthorizationGrant` should get helper methods for common things. Maybe you need this path for something user-defined (can such a thing exist?) or something, but the common cases should be obvious.

Reply:
In the latest commit there is now a helper method in `OAuthAuthorizationGrant` to get specific values for keys defined in `GrantConstants`.